docs/docs/workflows/troubleshooting/README.md

```diff
@@ -0,0 +1,60 @@
+# Troubleshooting Workflows
+
+Pipedream's unique build, test and deploy streamline your development process, but there might be times you run into issues with serverless infrastructure. Below are some common reasons why a workflow might not be behaving as you'd expect compared to locally run code or code hosted on traditional hosting.
+
+## `Pipedream Internal Error`
+
+A `Pipedream Internal Error` is thrown whenever there's an exception during the building or executing of a workflow that's outside the scope of the code for the individual components (steps or actions).
+
+There are a few known ways this can be caused and how to solve them.
+
+### Out of date actions or sources
+
+Pipedream components are updated continously.  But when new versions of actions and sources are published to the Pipedream Component Registry, your workflows are not updated by default.
+
+[An **Update** prompt](/workflows/steps/actions/#updating-actions-to-the-latest-version) is shown in the in the top right of the action if the component has a new version available.
+
+Sources do not feature an update button at this time, to receive the latest version, you'll need to create a new source, then attach it to your workflow.
+
+### New package versions issues
+
+If an NPM or PyPI package throws an error during either the building of the workflow or during it's execution, it may cause a `Pipedream Internal Error`.
+
+By default, Pipedream automatically updates NPM and PyPI packages to the latest version available. This is designed to make sure your workflows receive the latest package updates automatically.
+
+However, if a new package version includes a fatal bug, or changes it's export signature, then this may cause a `Pipedream Internal Error`.
```

Comment:
```suggestion
However, if a new package version introduces a bug, or changes it's export signature, then this may cause a `Pipedream Internal Error`.
```